open-sphere-plugins/filter-builder/src/main/java/io/opensphere/filterbuilder2/editor/FilterEditorPanel.java

```diff
@@ -128,15 +128,35 @@ private static void updateColumns(Filter filter, DataTypeInfo dataType)
         {
             return null;
         }
+
+        List<String> columns;
         List<String> keys = meta.getKeyNames();
         if (keys.size() > 0)
         {
-            return new ArrayList<>(keys);
+            columns = new ArrayList<>(keys);
         }
         else
         {
-            return new ArrayList<>(meta.getOriginalKeyNames());
+            columns = new ArrayList<>(meta.getOriginalKeyNames());
         }
+
+        // Remove fields that don't make sense in filter builder
+        String key;
+        if ((key = meta.getLatitudeKey()) != null)
```

Comment:
ACK. Don't assign to key in the if statement. I'd rather call meta.getLatitudeKey() twice. Looking at it's implementation, it's a map lookup, so it's not expensive. 3 instances of this.